src/main.c

```diff
@@ -195,46 +182,31 @@ int main(void)
 		LOG_ERR("Unable to configure LED for Golioth Logo");
 	}
 
-	/* Initialize app state */
-	app_state_init(client);
-
-	/* Initialize app work */
-	app_work_init(client);
-
-	/* Initialize DFU components */
-	app_dfu_init(client);
-
-	/* Initialize app settings */
-	app_settings_init(client);
-
-	/* Initialize app RPC */
-	app_rpc_init(client);
-
-	/* Register Golioth on_connect callback */
-	client->on_connect = golioth_on_connect;
-
-	/* Start LTE asynchronously if the nRF9160 is used
+#ifdef CONFIG_SOC_NRF9160
+	/* Start LTE asynchronously if the nRF9160 is used.
 	 * Golioth Client will start automatically when LTE connects
 	 */
-	IF_ENABLED(CONFIG_SOC_NRF9160, (LOG_INF("Connecting to LTE, this may take some time...");
-					lte_lc_init_and_connect_async(lte_handler);));
 
+	LOG_INF("Connecting to LTE, this may take some time...");
+	lte_lc_init_and_connect_async(lte_handler);
+
+#else
 	/* If nRF9160 is not used, start the Golioth Client and block until connected */
-	if (!IS_ENABLED(CONFIG_SOC_NRF9160)) {
-		/* Run WiFi/DHCP if necessary */
-		if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLES_COMMON)) {
-			net_connect();
-		}
 
-		/* Start Golioth client */
-		golioth_system_client_start();
+	/* Run WiFi/DHCP if necessary */
+	if (IS_ENABLED(CONFIG_GOLIOTH_SAMPLES_COMMON)) {
+		net_connect();
+	}
 
-		/* Block until connected to Golioth */
-		k_sem_take(&connected, K_FOREVER);
+	/* Start Golioth client */
+	start_golioth_client();
 
-		/* Turn on Golioth logo LED once connected */
-		gpio_pin_set_dt(&golioth_led, 1);
-	}
+	/* Block until connected to Golioth */
+	k_sem_take(&connected, K_FOREVER);
+
+	/* Turn on Golioth logo LED once connected */
+	gpio_pin_set_dt(&golioth_led, 1);
+#endif /* CONFIG_SOC_NRF9160 */
```

Comment:
Should this line be moved up before blocking on the semaphore?

i.e.

```c
#endif /* CONFIG_SOC_NRF9160 */

	/* Block until connected to Golioth */
	k_sem_take(&connected, K_FOREVER);
```

The way it is currently, we're not blocking on the semaphore (and we don't turn on the `golioth_led`) in the nRF9160 case.

Reply:
This is the expected behavior

- If we're not using an nRF9160 we block until the connection is available
- When using the nRF9160 we start the client asynchronously (no blocking) and use the LTE handler callback to start the client

But, you have found unnecessary code. The `gpio_pin_set_dt()` can be removed because it's handled in the `on_client_event()` callback.

Reply:
> This is the expected behavior

Got it 👍 what's the reason for only blocking in the non-nRF9160 case?

Reply:
If I remember correctly, the other network types will block (via `net_connect()`) until they're connected anyway so we might as well block until Golioth is connected. The nRF91 supports async, and since we use these for conference demos we want sensor readings displayed on Ostentus even if we don't have a network connection.